packages/core/src/domain/trackingConsent.ts

```diff
@@ -9,25 +13,36 @@ export type TrackingConsent = (typeof TrackingConsent)[keyof typeof TrackingCons
 export interface TrackingConsentState {
   tryToInit: (trackingConsent: TrackingConsent) => void
   update: (trackingConsent: TrackingConsent) => void
-  isGranted: () => boolean
+  isGranted: (startTime?: RelativeTime) => boolean
   observable: Observable<void>
 }
 
 export function createTrackingConsentState(currentConsent?: TrackingConsent): TrackingConsentState {
   const observable = new Observable<void>()
+  // Reuse the same expire delay as the session timeout delay because the tracking consent is expected to be valid for the same duration as the session.
+  const history = createValueHistory<TrackingConsent>({ expireDelay: SESSION_TIME_OUT_DELAY })
+
+  if (currentConsent) {
+    history.add(currentConsent, relativeNow())
+  }
 
   return {
     tryToInit(trackingConsent: TrackingConsent) {
       if (!currentConsent) {
         currentConsent = trackingConsent
+        history.add(trackingConsent, relativeNow())
       }
     },
     update(trackingConsent: TrackingConsent) {
       currentConsent = trackingConsent
```

Comment:
Now that you use the history, you can probably drop the `currentConsent` update and rename the parameter in `createTrackingConsentState` to `initialConsent`. Wdyt?